File: sdk/approute-sdk/python/approute/models/funds.py
from datetime import datetime

from approute.models._base import _Base


class FundingMethod(_Base):
    code: str
    name: str
    min_amount: float
    commission: float
    address: str
    ttl_minutes: int
    confirmations_required: int


class FundingMethodsResponse(_Base):
    items: list[FundingMethod]


class FundingInvoice(_Base):
    id: str
    method_code: str
    amount_expected: float
    commission: float
    credited: float
    memo_tag: str | None = None
    address: str
    tx_hash: str | None = None
    status: str
    confirmations_required: int | None = None
    confirmations: int | None = None
    created_at: datetime
    expires_at: datetime
    direction: str = "incoming"


class FundingInvoiceList(_Base):
    items: list[FundingInvoice]
    total: int


class FundingInvoiceCreateRequest(_Base):
    method_code: str
    amount: float


class FundingInvoiceTimeLeft(_Base):
    invoice_id: str
    expires_at: datetime
    seconds_left: int
    expired: bool


class TonDepositState(_Base):
    address: str
    memo_tag: str


class BybitState(_Base):
    recipient_uid: str
    linked: bool
    your_uid: str | None = None


class BybitAttachRequest(_Base):
    uid: str

File: sdk/approute-sdk/golang/model/funds.go
package model

import (
	"time"

	"github.com/approute/public-api-sdk-go/enum"
)

// FundingMethod describes one available deposit method.
type FundingMethod struct {
	Code                  enum.FundingMethodCode `json:"code"`
	Name                  string                 `json:"name"`
	MinAmount             float64                `json:"minAmount"`
	Commission            float64                `json:"commission"`
	Address               string                 `json:"address"`
	TTLMinutes            int                    `json:"ttlMinutes"`
	ConfirmationsRequired int                    `json:"confirmationsRequired"`
}

// FundingMethodsResponse is the list returned by GET /funds/methods.
type FundingMethodsResponse struct {
	Items []FundingMethod `json:"items"`
}

// FundingInvoice represents a single funding invoice.
type FundingInvoice struct {
	ID                    string                 `json:"id"`
	MethodCode            enum.FundingMethodCode `json:"methodCode"`
	AmountExpected        float64                `json:"amountExpected"`
	Commission            float64                `json:"commission"`
	Credited              float64                `json:"credited"`
	MemoTag               *string                `json:"memoTag,omitempty"`
	Address               string                 `json:"address"`
	TxHash                *string                `json:"txHash,omitempty"`
	Status                enum.FundingStatus     `json:"status"`
	ConfirmationsRequired *int                   `json:"confirmationsRequired,omitempty"`
	Confirmations         *int                   `json:"confirmations,omitempty"`
	CreatedAt             time.Time              `json:"createdAt"`
	ExpiresAt             time.Time              `json:"expiresAt"`
	Direction             string                 `json:"direction,omitempty"`
}

// FundingInvoiceList is the paginated response for GET /funds/invoices.
type FundingInvoiceList struct {
	Items []FundingInvoice `json:"items"`
	Total int              `json:"total"`
}

// FundingInvoiceCreateRequest is the body for POST /funds/invoices.
type FundingInvoiceCreateRequest struct {
	MethodCode enum.FundingMethodCode `json:"methodCode"`
	Amount     float64               `json:"amount"`
}

// FundingInvoiceTimeLeft reports how much time remains before an invoice expires.
type FundingInvoiceTimeLeft struct {
	InvoiceID   string    `json:"invoiceId"`
	ExpiresAt   time.Time `json:"expiresAt"`
	SecondsLeft int       `json:"secondsLeft"`
	Expired     bool      `json:"expired"`
}

// TonDepositState holds the TON deposit address and memo tag.
type TonDepositState struct {
	Address string `json:"address"`
	MemoTag string `json:"memoTag"`
}

// BybitState reports the Bybit UID linkage state.
type BybitState struct {
	RecipientUID string  `json:"recipientUid"`
	Linked       bool    `json:"linked"`
	YourUID      *string `json:"yourUid,omitempty"`
}

// BybitAttachRequest is the body for POST /funds/bybit/attach.
type BybitAttachRequest struct {
	UID string `json:"uid"`
}
